// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/confirm-sign-up-component/confirm-sign-up.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../../../providers/amplify.service';

function ConfirmSignUpComponentIonic_div_1_div_52_Template(rf, ctx) { if (rf & 1) {
    var _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 16);
    ɵngcc0.ɵɵtext(1, "\n    ");
    ɵngcc0.ɵɵelementStart(2, "div", 17);
    ɵngcc0.ɵɵtext(3, "\n      ");
    ɵngcc0.ɵɵelementStart(4, "span", 18);
    ɵngcc0.ɵɵtext(5, "\u26A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n      ");
    ɵngcc0.ɵɵelementStart(7, "div", 19);
    ɵngcc0.ɵɵtext(8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(9, "\n      ");
    ɵngcc0.ɵɵelementStart(10, "a", 20);
    ɵngcc0.ɵɵlistener("click", function ConfirmSignUpComponentIonic_div_1_div_52_Template_a_click_10_listener() { ɵngcc0.ɵɵrestoreView(_r4); var ctx_r3 = ɵngcc0.ɵɵnextContext(2); return ctx_r3.onAlertClose(); });
    ɵngcc0.ɵɵtext(11, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(12, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(13, "\n  ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(8);
    ɵngcc0.ɵɵtextInterpolate(ctx_r2.amplifyService.i18n().get(ctx_r2.errorMessage));
} }
function ConfirmSignUpComponentIonic_div_1_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵtext(1, "\n  ");
    ɵngcc0.ɵɵelementStart(2, "div", 2);
    ɵngcc0.ɵɵtext(3, "\n    ");
    ɵngcc0.ɵɵelementStart(4, "div", 3);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n    ");
    ɵngcc0.ɵɵelementStart(7, "ion-list");
    ɵngcc0.ɵɵtext(8, "\n      ");
    ɵngcc0.ɵɵelementStart(9, "ion-item", 4);
    ɵngcc0.ɵɵtext(10, "\n        ");
    ɵngcc0.ɵɵelementStart(11, "ion-label", 5);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(13, "\n        ");
    ɵngcc0.ɵɵelementStart(14, "ion-input", 6);
    ɵngcc0.ɵɵlistener("input", function ConfirmSignUpComponentIonic_div_1_Template_ion_input_input_14_listener($event) { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.setUsername($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(15, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(16, "\n\n      ");
    ɵngcc0.ɵɵelementStart(17, "ion-item", 4);
    ɵngcc0.ɵɵtext(18, "\n        ");
    ɵngcc0.ɵɵelementStart(19, "ion-label", 5);
    ɵngcc0.ɵɵtext(20);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(21, "\n        ");
    ɵngcc0.ɵɵelementStart(22, "ion-input", 7, 8);
    ɵngcc0.ɵɵlistener("input", function ConfirmSignUpComponentIonic_div_1_Template_ion_input_input_22_listener() { ɵngcc0.ɵɵrestoreView(_r6); var _r1 = ɵngcc0.ɵɵreference(23); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.setCode(_r1.value); })("keyup.enter", function ConfirmSignUpComponentIonic_div_1_Template_ion_input_keyup_enter_22_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.onConfirm(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(24, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(25, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(26, "\n\n    ");
    ɵngcc0.ɵɵelementStart(27, "div", 9);
    ɵngcc0.ɵɵtext(28, "\n      ");
    ɵngcc0.ɵɵelementStart(29, "div");
    ɵngcc0.ɵɵtext(30, "\n        ");
    ɵngcc0.ɵɵelementStart(31, "ion-button", 10);
    ɵngcc0.ɵɵlistener("click", function ConfirmSignUpComponentIonic_div_1_Template_ion_button_click_31_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r9 = ɵngcc0.ɵɵnextContext(); return ctx_r9.onConfirm(); });
    ɵngcc0.ɵɵtext(32);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(33, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(34, "\n    ");
    ɵngcc0.ɵɵelementStart(35, "div", 11);
    ɵngcc0.ɵɵtext(36, "\n      ");
    ɵngcc0.ɵɵelementStart(37, "div", 12);
    ɵngcc0.ɵɵtext(38);
    ɵngcc0.ɵɵelementStart(39, "a", 13);
    ɵngcc0.ɵɵlistener("click", function ConfirmSignUpComponentIonic_div_1_Template_a_click_39_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.onSignIn(); });
    ɵngcc0.ɵɵtext(40);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(41, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(42, "\n      ");
    ɵngcc0.ɵɵelementStart(43, "div", 12);
    ɵngcc0.ɵɵtext(44);
    ɵngcc0.ɵɵelementStart(45, "a", 14);
    ɵngcc0.ɵɵlistener("click", function ConfirmSignUpComponentIonic_div_1_Template_a_click_45_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r11 = ɵngcc0.ɵɵnextContext(); return ctx_r11.onResend(); });
    ɵngcc0.ɵɵtext(46);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(47, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(48, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(49, "\n  ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(50, "\n");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(51, "\n\n  ");
    ɵngcc0.ɵɵtemplate(52, ConfirmSignUpComponentIonic_div_1_div_52_Template, 14, 1, "div", 15);
    ɵngcc0.ɵɵtext(53, "\n\n");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate1("\n      ", ctx_r0.amplifyService.i18n().get("Confirm your sign up code"), "\n    ");
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get(ctx_r0.getUsernameLabel()), "\n        ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("value", ctx_r0.username);
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Code *"), "\n        ");
    ɵngcc0.ɵɵadvance(12);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Confirm Code"), "");
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵtextInterpolate1("\n        ", ctx_r0.amplifyService.i18n().get("Have an account?"), "\n        ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Sign In"), "\n        ");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1("\n        ", ctx_r0.amplifyService.i18n().get("Lost your code?"), "\n        ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Resend"), "\n        ");
    ɵngcc0.ɵɵadvance(6);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.errorMessage);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { ConfirmSignUpComponentCore } from './confirm-sign-up.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div\n  class=\"amplify-authenticator amplify-authenticator-ionic\"\n  *ngIf=\"_show\"\n  data-test=\"" + auth.confirmSignUp.section + "\"\n  >\n  <div class=\"amplify-form-body\" data-test=\"" + auth.confirmSignUp.bodySection + "\">\n    <div\n      class=\"amplify-form-header amplify-form-header-ionic\"\n      data-test=\"" + auth.confirmSignUp.headerSection + "\"\n      >\n      {{ this.amplifyService.i18n().get('Confirm your sign up code') }}\n    </div>\n    <ion-list>\n      <ion-item lines=\"none\">\n        <ion-label class=\"amplify-input-label amplify-input-label-ionic\" position=\"stacked\">\n          {{ this.amplifyService.i18n().get(getUsernameLabel()) }}\n        </ion-label>\n        <ion-input type=\"text\"\n          class=\"amplify-form-input\"\n          (input)=\"setUsername($event.target.value)\"\n          [value]=\"username\"\n          data-test=\"" + auth.confirmSignUp.usernameInput + "\"\n        ></ion-input>\n      </ion-item>\n\n      <ion-item lines=\"none\">\n        <ion-label class=\"amplify-input-label amplify-input-label-ionic\" position=\"stacked\">\n          {{ this.amplifyService.i18n().get('Code *') }}\n        </ion-label>\n        <ion-input\n          #code\n          type=\"text\"\n          class=\"amplify-form-input\"\n          (input)=\"setCode(code.value)\"\n          (keyup.enter)=\"onConfirm()\"\n          data-test=\"" + auth.confirmSignUp.confirmationCodeInput + "\"\n        ></ion-input>\n      </ion-item>\n    </ion-list>\n\n    <div class=\"amplify-form-actions\">\n      <div>\n        <ion-button\n          expand=\"block\"\n          color=\"primary\"\n          (click)=\"onConfirm()\"\n          data-test=\"" + auth.confirmSignUp.confirmButton + "\"\n          >\n          {{ this.amplifyService.i18n().get('Confirm Code') }}</ion-button>\n      </div>\n    <div class=\"amplify-form-row\">\n      <div class=\"amplify-form-signup\">\n        {{ this.amplifyService.i18n().get('Have an account?') }}\n        <a\n          class=\"amplify-form-link\"\n          (click)=\"onSignIn()\"\n          data-test=\"" + auth.confirmSignUp.backToSignInLink + "\"\n          >\n          {{ this.amplifyService.i18n().get('Sign In') }}\n        </a>\n      </div>\n      <div class=\"amplify-form-signup\">\n        {{ this.amplifyService.i18n().get('Lost your code?') }}\n        <a\n          class=\"amplify-form-link\"\n          (click)=\"onResend()\"\n          data-test=\"" + auth.confirmSignUp.resendCodeLink + "\"\n          >\n          {{ this.amplifyService.i18n().get('Resend') }}\n        </a>\n      </div>\n    </div>\n  </div>\n</div>\n\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ this.amplifyService.i18n().get(errorMessage) }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n\n</div>\n";
var ConfirmSignUpComponentIonic = /** @class */ (function (_super) {
    __extends(ConfirmSignUpComponentIonic, _super);
    function ConfirmSignUpComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    ConfirmSignUpComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
ConfirmSignUpComponentIonic.ɵfac = function ConfirmSignUpComponentIonic_Factory(t) { return new (t || ConfirmSignUpComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
ConfirmSignUpComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ConfirmSignUpComponentIonic, selectors: [["amplify-auth-confirm-sign-up-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [["class", "amplify-authenticator amplify-authenticator-ionic", "data-test", "confirm-sign-up-section", 4, "ngIf"], ["data-test", "confirm-sign-up-section", 1, "amplify-authenticator", "amplify-authenticator-ionic"], ["data-test", "confirm-sign-up-body-section", 1, "amplify-form-body"], ["data-test", "confirm-sign-up-header-section", 1, "amplify-form-header", "amplify-form-header-ionic"], ["lines", "none"], ["position", "stacked", 1, "amplify-input-label", "amplify-input-label-ionic"], ["type", "text", "data-test", "confirm-sign-up-username-input", 1, "amplify-form-input", 3, "value", "input"], ["type", "text", "data-test", "confirm-sign-up-confirmation-code-input", 1, "amplify-form-input", 3, "input", "keyup.enter"], ["code", ""], [1, "amplify-form-actions"], ["expand", "block", "color", "primary", "data-test", "confirm-sign-up-confirm-button", 3, "click"], [1, "amplify-form-row"], [1, "amplify-form-signup"], ["data-test", "confirm-sign-up-back-to-sign-in-link", 1, "amplify-form-link", 3, "click"], ["data-test", "confirm-sign-up-resend-code-link", 1, "amplify-form-link", 3, "click"], ["class", "amplify-alert", 4, "ngIf"], [1, "amplify-alert"], [1, "amplify-alert-body"], [1, "amplify-alert-icon"], [1, "amplify-alert-message"], [1, "amplify-alert-close", 3, "click"]], template: function ConfirmSignUpComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtext(0, "\n");
        ɵngcc0.ɵɵtemplate(1, ConfirmSignUpComponentIonic_div_1_Template, 54, 10, "div", 0);
        ɵngcc0.ɵɵtext(2, "\n");
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx._show);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ConfirmSignUpComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-confirm-sign-up-ionic',
                template: template
            }]
    }], function () { return [{ type: ɵngcc2.AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return ConfirmSignUpComponentIonic;
}(ConfirmSignUpComponentCore));
export { ConfirmSignUpComponentIonic };
