// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/require-new-password-component/require-new-password.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../../../providers/amplify.service';

function RequireNewPasswordComponentIonic_div_1_div_35_Template(rf, ctx) { if (rf & 1) {
    var _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 13);
    ɵngcc0.ɵɵtext(1, "\n    ");
    ɵngcc0.ɵɵelementStart(2, "div", 14);
    ɵngcc0.ɵɵtext(3, "\n      ");
    ɵngcc0.ɵɵelementStart(4, "span", 15);
    ɵngcc0.ɵɵtext(5, "\u26A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n      ");
    ɵngcc0.ɵɵelementStart(7, "div", 16);
    ɵngcc0.ɵɵtext(8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(9, "\n      ");
    ɵngcc0.ɵɵelementStart(10, "a", 17);
    ɵngcc0.ɵɵlistener("click", function RequireNewPasswordComponentIonic_div_1_div_35_Template_a_click_10_listener() { ɵngcc0.ɵɵrestoreView(_r4); var ctx_r3 = ɵngcc0.ɵɵnextContext(2); return ctx_r3.onAlertClose(); });
    ɵngcc0.ɵɵtext(11, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(12, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(13, "\n  ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(8);
    ɵngcc0.ɵɵtextInterpolate(ctx_r2.errorMessage);
} }
function RequireNewPasswordComponentIonic_div_1_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵtext(1, "\n  ");
    ɵngcc0.ɵɵelementStart(2, "div", 2);
    ɵngcc0.ɵɵtext(3, "\n    ");
    ɵngcc0.ɵɵelementStart(4, "div", 3);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n    ");
    ɵngcc0.ɵɵelementStart(7, "ion-list");
    ɵngcc0.ɵɵtext(8, "\n      ");
    ɵngcc0.ɵɵelementStart(9, "ion-item", 4);
    ɵngcc0.ɵɵtext(10, "\n        ");
    ɵngcc0.ɵɵelementStart(11, "ion-label", 5);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(13, "\n        ");
    ɵngcc0.ɵɵelementStart(14, "ion-input", 6, 7);
    ɵngcc0.ɵɵlistener("input", function RequireNewPasswordComponentIonic_div_1_Template_ion_input_input_14_listener() { ɵngcc0.ɵɵrestoreView(_r6); var _r1 = ɵngcc0.ɵɵreference(15); var ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.setPassword(_r1.value); })("keyup.enter", function RequireNewPasswordComponentIonic_div_1_Template_ion_input_keyup_enter_14_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r7 = ɵngcc0.ɵɵnextContext(); return ctx_r7.onSubmit(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(16, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(17, "\n\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(18, "\n\n    ");
    ɵngcc0.ɵɵelementStart(19, "div", 8);
    ɵngcc0.ɵɵtext(20, "\n      ");
    ɵngcc0.ɵɵelementStart(21, "div", 9);
    ɵngcc0.ɵɵtext(22, "\n        ");
    ɵngcc0.ɵɵelementStart(23, "ion-button", 10);
    ɵngcc0.ɵɵlistener("click", function RequireNewPasswordComponentIonic_div_1_Template_ion_button_click_23_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.onSignIn(); });
    ɵngcc0.ɵɵtext(24);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(25, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(26, "\n      ");
    ɵngcc0.ɵɵelementStart(27, "div", 9);
    ɵngcc0.ɵɵtext(28, "\n        ");
    ɵngcc0.ɵɵelementStart(29, "ion-button", 11);
    ɵngcc0.ɵɵlistener("click", function RequireNewPasswordComponentIonic_div_1_Template_ion_button_click_29_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r9 = ɵngcc0.ɵɵnextContext(); return ctx_r9.onSubmit(); });
    ɵngcc0.ɵɵtext(30, "Submit");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(31, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(32, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(33, "\n  ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(34, "\n  ");
    ɵngcc0.ɵɵtemplate(35, RequireNewPasswordComponentIonic_div_1_div_35_Template, 14, 1, "div", 12);
    ɵngcc0.ɵɵtext(36, "\n");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate1("\n      ", ctx_r0.amplifyService.i18n().get("Reset your password"), "\n    ");
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Password"), "\n        ");
    ɵngcc0.ɵɵadvance(12);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.amplifyService.i18n().get("Back to Sign In"));
    ɵngcc0.ɵɵadvance(11);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.errorMessage);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { RequireNewPasswordComponentCore } from './require-new-password.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div\n  class=\"amplify-authenticator amplify-authenticator-ionic\"\n  *ngIf=\"_show\"\n  data-test=\"" + auth.requireNewPassword.section + "\"\n  >\n  <div class=\"amplify-form-body\" data-test=\"" + auth.requireNewPassword.bodySection + "\">\n    <div\n      class=\"amplify-form-header amplify-form-header-ionic\"\n      data-test=\"" + auth.requireNewPassword.headerSection + "\"\n      >\n      {{ this.amplifyService.i18n().get('Reset your password') }}\n    </div>\n    <ion-list>\n      <ion-item lines=\"none\">\n        <ion-label class=\"amplify-input-label amplify-input-label-ionic\" position=\"stacked\">\n          {{ this.amplifyService.i18n().get('Password') }}\n        </ion-label>\n        <ion-input\n          #password\n          type=\"password\"\n          class=\"amplify-form-input\"\n          (input)=\"setPassword(password.value)\"\n          (keyup.enter)=\"onSubmit()\"\n          data-test=\"" + auth.requireNewPassword.newPasswordInput + "\"\n        ></ion-input>\n      </ion-item>\n\n    </ion-list>\n\n    <div class=\"amplify-form-actions\">\n      <div class=\"amplify-form-row\">\n        <ion-button\n          expand=\"block\"\n          (click)=\"onSignIn()\"\n          data-test=\"" + auth.requireNewPassword.backToSignInLink + "\"\n        >{{ this.amplifyService.i18n().get('Back to Sign In') }}</ion-button>\n      </div>\n      <div class=\"amplify-form-row\">\n        <ion-button\n          expand=\"block\"\n          (click)=\"onSubmit()\"\n          data-test=\"" + auth.requireNewPassword.submitButton + "\"\n        >Submit</ion-button>\n      </div>\n    </div>\n  </div>\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n</div>\n";
var RequireNewPasswordComponentIonic = /** @class */ (function (_super) {
    __extends(RequireNewPasswordComponentIonic, _super);
    function RequireNewPasswordComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    RequireNewPasswordComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
RequireNewPasswordComponentIonic.ɵfac = function RequireNewPasswordComponentIonic_Factory(t) { return new (t || RequireNewPasswordComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
RequireNewPasswordComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: RequireNewPasswordComponentIonic, selectors: [["amplify-auth-require-new-password-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [["class", "amplify-authenticator amplify-authenticator-ionic", "data-test", "require-new-password-section", 4, "ngIf"], ["data-test", "require-new-password-section", 1, "amplify-authenticator", "amplify-authenticator-ionic"], ["data-test", "require-new-password-body-section", 1, "amplify-form-body"], ["data-test", "require-new-password-header-section", 1, "amplify-form-header", "amplify-form-header-ionic"], ["lines", "none"], ["position", "stacked", 1, "amplify-input-label", "amplify-input-label-ionic"], ["type", "password", "data-test", "require-new-password-new-password-input", 1, "amplify-form-input", 3, "input", "keyup.enter"], ["password", ""], [1, "amplify-form-actions"], [1, "amplify-form-row"], ["expand", "block", "data-test", "require-new-password-back-to-sign-in-link", 3, "click"], ["expand", "block", "data-test", "require-new-password-submit-button", 3, "click"], ["class", "amplify-alert", 4, "ngIf"], [1, "amplify-alert"], [1, "amplify-alert-body"], [1, "amplify-alert-icon"], [1, "amplify-alert-message"], [1, "amplify-alert-close", 3, "click"]], template: function RequireNewPasswordComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtext(0, "\n");
        ɵngcc0.ɵɵtemplate(1, RequireNewPasswordComponentIonic_div_1_Template, 37, 4, "div", 0);
        ɵngcc0.ɵɵtext(2, "\n");
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx._show);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RequireNewPasswordComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-require-new-password-ionic',
                template: template
            }]
    }], function () { return [{ type: ɵngcc2.AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return RequireNewPasswordComponentIonic;
}(RequireNewPasswordComponentCore));
export { RequireNewPasswordComponentIonic };
