// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/greeting-component/greeting.factory.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, ViewChild, ComponentFactoryResolver, } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { GreetingComponentIonic } from './greeting.component.ionic';
import { GreetingComponentCore } from './greeting.component.core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../directives/dynamic.component.directive';

function GreetingComponent_ng_template_3_Template(rf, ctx) { }
var GreetingComponent = /** @class */ (function () {
    function GreetingComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.usernameAttributes = 'username';
    }
    GreetingComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    GreetingComponent.prototype.ngOnDestroy = function () { };
    GreetingComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic'
            ? new ComponentMount(GreetingComponentIonic, {
                authState: this.authState,
                usernameAttributes: this.usernameAttributes,
            })
            : new ComponentMount(GreetingComponentCore, {
                authState: this.authState,
                usernameAttributes: this.usernameAttributes,
            });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    /** @nocollapse */
    GreetingComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    GreetingComponent.propDecorators = {
        "framework": [{ type: Input },],
        "authState": [{ type: Input },],
        "usernameAttributes": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
GreetingComponent.ɵfac = function GreetingComponent_Factory(t) { return new (t || GreetingComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
GreetingComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GreetingComponent, selectors: [["amplify-auth-greetings"]], viewQuery: function GreetingComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DynamicComponentDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.componentHost = _t.first);
    } }, inputs: { usernameAttributes: "usernameAttributes", framework: "framework", authState: "authState" }, decls: 6, vars: 0, consts: [["component-host", ""]], template: function GreetingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtext(0, "\n\t\t");
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵtext(2, "\n\t\t\t");
        ɵngcc0.ɵɵtemplate(3, GreetingComponent_ng_template_3_Template, 0, 0, "ng-template", 0);
        ɵngcc0.ɵɵtext(4, "\n\t\t");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(5, "\n\t");
    } }, directives: [ɵngcc1.DynamicComponentDirective], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GreetingComponent, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-greetings',
                template: "\n\t\t<div>\n\t\t\t<ng-template component-host></ng-template>\n\t\t</div>\n\t"
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }]; }, { usernameAttributes: [{
            type: Input
        }], framework: [{
            type: Input
        }], authState: [{
            type: Input
        }], componentHost: [{
            type: ViewChild,
            args: [DynamicComponentDirective]
        }] }); })();
    return GreetingComponent;
}());
export { GreetingComponent };
