// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/sign-in-component/sign-in.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../username-field-component/username-field.component.ionic';
import * as ɵngcc3 from '../../../providers/amplify.service';

function SignInComponentIonic_div_1_div_31_Template(rf, ctx) { if (rf & 1) {
    var _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 14);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementStart(2, "a", 15);
    ɵngcc0.ɵɵlistener("click", function SignInComponentIonic_div_1_div_31_Template_a_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r6); var ctx_r5 = ɵngcc0.ɵɵnextContext(2); return ctx_r5.onSignUp(); });
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(4, "\n        ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r2.amplifyService.i18n().get("No account?"), "\n          ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("\n            ", ctx_r2.amplifyService.i18n().get("Create account"), "\n          ");
} }
function SignInComponentIonic_div_1_div_33_Template(rf, ctx) { if (rf & 1) {
    var _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 14);
    ɵngcc0.ɵɵtext(1, "\n          ");
    ɵngcc0.ɵɵelementStart(2, "a", 16);
    ɵngcc0.ɵɵlistener("click", function SignInComponentIonic_div_1_div_33_Template_a_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r8); var ctx_r7 = ɵngcc0.ɵɵnextContext(2); return ctx_r7.onForgotPassword(); });
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(4, "\n        ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1("\n            ", ctx_r3.amplifyService.i18n().get("Reset Password"), "\n          ");
} }
function SignInComponentIonic_div_1_div_38_Template(rf, ctx) { if (rf & 1) {
    var _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 17);
    ɵngcc0.ɵɵtext(1, "\n    ");
    ɵngcc0.ɵɵelementStart(2, "div", 18);
    ɵngcc0.ɵɵtext(3, "\n      ");
    ɵngcc0.ɵɵelementStart(4, "span", 19);
    ɵngcc0.ɵɵtext(5, "\u26A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n      ");
    ɵngcc0.ɵɵelementStart(7, "div", 20);
    ɵngcc0.ɵɵtext(8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(9, "\n      ");
    ɵngcc0.ɵɵelementStart(10, "a", 21);
    ɵngcc0.ɵɵlistener("click", function SignInComponentIonic_div_1_div_38_Template_a_click_10_listener() { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r9 = ɵngcc0.ɵɵnextContext(2); return ctx_r9.onAlertClose(); });
    ɵngcc0.ɵɵtext(11, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(12, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(13, "\n  ");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(8);
    ɵngcc0.ɵɵtextInterpolate(ctx_r4.amplifyService.i18n().get(ctx_r4.errorMessage));
} }
function SignInComponentIonic_div_1_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵtext(1, "\n  ");
    ɵngcc0.ɵɵelementStart(2, "div", 2);
    ɵngcc0.ɵɵtext(3, "\n    ");
    ɵngcc0.ɵɵelementStart(4, "div", 3);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(6, "\n    ");
    ɵngcc0.ɵɵelementStart(7, "ion-list", 4);
    ɵngcc0.ɵɵtext(8, "\n      ");
    ɵngcc0.ɵɵelementStart(9, "amplify-auth-username-field-ionic", 5);
    ɵngcc0.ɵɵlistener("usernameFieldChanged", function SignInComponentIonic_div_1_Template_amplify_auth_username_field_ionic_usernameFieldChanged_9_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r11 = ɵngcc0.ɵɵnextContext(); return ctx_r11.onUsernameFieldChanged($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(10, "\n      ");
    ɵngcc0.ɵɵelementStart(11, "ion-item", 4);
    ɵngcc0.ɵɵtext(12, "\n        ");
    ɵngcc0.ɵɵelementStart(13, "ion-label", 6);
    ɵngcc0.ɵɵtext(14);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(15, "\n        ");
    ɵngcc0.ɵɵelementStart(16, "ion-input", 7, 8);
    ɵngcc0.ɵɵlistener("input", function SignInComponentIonic_div_1_Template_ion_input_input_16_listener() { ɵngcc0.ɵɵrestoreView(_r12); var _r1 = ɵngcc0.ɵɵreference(17); var ctx_r13 = ɵngcc0.ɵɵnextContext(); return ctx_r13.setPassword(_r1.value); })("keyup.enter", function SignInComponentIonic_div_1_Template_ion_input_keyup_enter_16_listener() { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.onSignIn(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(18, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(19, "\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(20, "\n    ");
    ɵngcc0.ɵɵelementStart(21, "div", 9);
    ɵngcc0.ɵɵtext(22, "\n      ");
    ɵngcc0.ɵɵelementStart(23, "div", 10);
    ɵngcc0.ɵɵtext(24, "\n        ");
    ɵngcc0.ɵɵelementStart(25, "ion-button", 11);
    ɵngcc0.ɵɵlistener("click", function SignInComponentIonic_div_1_Template_ion_button_click_25_listener() { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r15 = ɵngcc0.ɵɵnextContext(); return ctx_r15.onSignIn(); });
    ɵngcc0.ɵɵtext(26);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(27, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(28, "\n\n      ");
    ɵngcc0.ɵɵelementStart(29, "div", 10);
    ɵngcc0.ɵɵtext(30, "\n        ");
    ɵngcc0.ɵɵtemplate(31, SignInComponentIonic_div_1_div_31_Template, 5, 2, "div", 12);
    ɵngcc0.ɵɵtext(32, "\n        ");
    ɵngcc0.ɵɵtemplate(33, SignInComponentIonic_div_1_div_33_Template, 5, 1, "div", 12);
    ɵngcc0.ɵɵtext(34, "\n      ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(35, "\n\n    ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(36, "\n  ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtext(37, "\n\n  ");
    ɵngcc0.ɵɵtemplate(38, SignInComponentIonic_div_1_div_38_Template, 14, 1, "div", 13);
    ɵngcc0.ɵɵtext(39, "\n");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate1("\n      ", ctx_r0.amplifyService.i18n().get("Sign in to your account"), "\n    ");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("usernameAttributes", ctx_r0._usernameAttributes);
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate1("\n          ", ctx_r0.amplifyService.i18n().get("Password *"), "\n        ");
    ɵngcc0.ɵɵadvance(12);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.amplifyService.i18n().get("Sign In"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.shouldHide("SignUp"));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.shouldHide("ForgotPassword"));
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.errorMessage);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { SignInComponentCore } from './sign-in.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div\n  class=\"amplify-authenticator\"\n  *ngIf=\"_show\"\n  data-test=\"" + auth.signIn.section + "\"\n  >\n  <div class=\"amplify-form-body\" data-test=\"" + auth.signIn.bodySection + "\">\n    <div class=\"amplify-form-header\" data-test=\"" + auth.signIn.headerSection + "\">\n      {{ this.amplifyService.i18n().get('Sign in to your account') }}\n    </div>\n    <ion-list lines=\"none\">\n      <amplify-auth-username-field-ionic\n        [usernameAttributes]=\"_usernameAttributes\"\n        (usernameFieldChanged)=\"onUsernameFieldChanged($event)\"\n      ></amplify-auth-username-field-ionic>\n      <ion-item lines=\"none\">\n        <ion-label class=\"amplify-input-label\" for=\"password\" position=\"stacked\">\n          {{ this.amplifyService.i18n().get('Password *') }}\n        </ion-label>\n        <ion-input\n          #password\n          type=\"password\"\n          class=\"amplify-form-input\"\n          (input)=\"setPassword(password.value)\"\n          (keyup.enter)=\"onSignIn()\"\n          data-test=\"" + auth.signIn.passwordInput + "\"\n        ></ion-input>\n      </ion-item>\n    </ion-list>\n    <div class=\"amplify-form-actions\">\n      <div class=\"amplify-form-row\">\n        <ion-button expand=\"block\" color=\"primary\"\n          (click)=\"onSignIn()\"\n          data-test=\"" + auth.signIn.signInButton + "\"\n          >{{ this.amplifyService.i18n().get('Sign In') }}</ion-button>\n      </div>\n\n      <div class=\"amplify-form-row\">\n        <div class=\"amplify-form-signup\" *ngIf=\"!shouldHide('SignUp')\">\n          {{ this.amplifyService.i18n().get('No account?') }}\n          <a\n            class=\"amplify-form-link\"\n            (click)=\"onSignUp()\"\n            data-test=\"" + auth.signIn.createAccountLink + "\"\n            >\n            {{ this.amplifyService.i18n().get('Create account') }}\n          </a>\n        </div>\n        <div class=\"amplify-form-signup\" *ngIf=\"!shouldHide('ForgotPassword')\">\n          <a class=\"amplify-form-link\" (click)=\"onForgotPassword()\" data-test=\"" + auth.signIn.forgotPasswordLink + "\">\n            {{ this.amplifyService.i18n().get('Reset Password') }}\n          </a>\n        </div>\n      </div>\n\n    </div>\n  </div>\n\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ this.amplifyService.i18n().get(errorMessage) }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n</div>\n";
var SignInComponentIonic = /** @class */ (function (_super) {
    __extends(SignInComponentIonic, _super);
    function SignInComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    SignInComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
SignInComponentIonic.ɵfac = function SignInComponentIonic_Factory(t) { return new (t || SignInComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
SignInComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SignInComponentIonic, selectors: [["amplify-auth-sign-in-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [["class", "amplify-authenticator", "data-test", "sign-in-section", 4, "ngIf"], ["data-test", "sign-in-section", 1, "amplify-authenticator"], ["data-test", "sign-in-body-section", 1, "amplify-form-body"], ["data-test", "sign-in-header-section", 1, "amplify-form-header"], ["lines", "none"], [3, "usernameAttributes", "usernameFieldChanged"], ["for", "password", "position", "stacked", 1, "amplify-input-label"], ["type", "password", "data-test", "sign-in-password-input", 1, "amplify-form-input", 3, "input", "keyup.enter"], ["password", ""], [1, "amplify-form-actions"], [1, "amplify-form-row"], ["expand", "block", "color", "primary", "data-test", "sign-in-sign-in-button", 3, "click"], ["class", "amplify-form-signup", 4, "ngIf"], ["class", "amplify-alert", 4, "ngIf"], [1, "amplify-form-signup"], ["data-test", "sign-in-create-account-link", 1, "amplify-form-link", 3, "click"], ["data-test", "sign-in-forgot-password-link", 1, "amplify-form-link", 3, "click"], [1, "amplify-alert"], [1, "amplify-alert-body"], [1, "amplify-alert-icon"], [1, "amplify-alert-message"], [1, "amplify-alert-close", 3, "click"]], template: function SignInComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtext(0, "\n");
        ɵngcc0.ɵɵtemplate(1, SignInComponentIonic_div_1_Template, 40, 7, "div", 0);
        ɵngcc0.ɵɵtext(2, "\n");
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx._show);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.UsernameFieldComponentIonic], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SignInComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-sign-in-ionic',
                template: template
            }]
    }], function () { return [{ type: ɵngcc3.AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return SignInComponentIonic;
}(SignInComponentCore));
export { SignInComponentIonic };
