// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/authenticator/authenticator.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../providers';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../sign-in-component/sign-in.component.ionic';
import * as ɵngcc4 from '../sign-up-component/sign-up.component.ionic';
import * as ɵngcc5 from '../confirm-sign-up-component/confirm-sign-up.component.ionic';
import * as ɵngcc6 from '../confirm-sign-in-component/confirm-sign-in-component.ionic';
import * as ɵngcc7 from '../forgot-password-component/forgot-password.component.ionic';
import * as ɵngcc8 from '../greeting-component/greeting.component.ionic';
import * as ɵngcc9 from '../require-new-password-component/require-new-password.component.ionic';

function AuthenticatorIonicComponent_amplify_auth_sign_in_ionic_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-sign-in-ionic", 5);
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r0.authState)("usernameAttributes", ctx_r0._usernameAttributes)("hide", ctx_r0.hide);
} }
function AuthenticatorIonicComponent_amplify_auth_sign_up_ionic_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-sign-up-ionic", 6);
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r1.authState)("signUpConfig", ctx_r1._signUpConfig)("usernameAttributes", ctx_r1._usernameAttributes)("hide", ctx_r1.hide);
} }
function AuthenticatorIonicComponent_amplify_auth_confirm_sign_up_ionic_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-confirm-sign-up-ionic", 5);
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r2.authState)("usernameAttributes", ctx_r2._usernameAttributes)("hide", ctx_r2.hide);
} }
function AuthenticatorIonicComponent_amplify_auth_confirm_sign_in_ionic_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-confirm-sign-in-ionic", 7);
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r3.authState)("hide", ctx_r3.hide);
} }
function AuthenticatorIonicComponent_amplify_auth_forgot_password_ionic_11_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-forgot-password-ionic", 5);
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r4.authState)("usernameAttributes", ctx_r4._usernameAttributes)("hide", ctx_r4.hide);
} }
function AuthenticatorIonicComponent_amplify_auth_greetings_ionic_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-greetings-ionic", 8);
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r5.authState)("usernameAttributes", ctx_r5._usernameAttributes);
} }
function AuthenticatorIonicComponent_amplify_auth_require_new_password_ionic_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-require-new-password-ionic", 7);
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r6.authState)("hide", ctx_r6.hide);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, } from '@angular/core';
import { AmplifyService } from '../../../providers';
import { AuthenticatorComponentCore } from './authenticator.component.core';
var template = "\n<div class=\"amplify-authenticator amplify-authenticator-ionic \">\n\n<amplify-auth-sign-in-ionic\n  *ngIf=\"!shouldHide('SignIn')\"\n  [authState]=\"authState\"\n  [usernameAttributes]=\"_usernameAttributes\"\n  [hide]=\"hide\"\n></amplify-auth-sign-in-ionic>\n\n<amplify-auth-sign-up-ionic\n  *ngIf=\"!shouldHide('SignUp')\"\n  [authState]=\"authState\"\n  [signUpConfig]=\"_signUpConfig\"\n  [usernameAttributes]=\"_usernameAttributes\"\n  [hide]=\"hide\"\n></amplify-auth-sign-up-ionic>\n\n<amplify-auth-confirm-sign-up-ionic\n  *ngIf=\"!shouldHide('ConfirmSignUp')\"\n  [authState]=\"authState\"\n  [usernameAttributes]=\"_usernameAttributes\"\n  [hide]=\"hide\"\n></amplify-auth-confirm-sign-up-ionic>\n\n<amplify-auth-confirm-sign-in-ionic\n  *ngIf=\"!shouldHide('ConfirmSignIn')\"\n  [authState]=\"authState\"\n  [hide]=\"hide\"\n></amplify-auth-confirm-sign-in-ionic>\n\n<amplify-auth-forgot-password-ionic\n  *ngIf=\"!shouldHide('ForgotPassword')\"\n  [authState]=\"authState\"\n  [usernameAttributes]=\"_usernameAttributes\"\n  [hide]=\"hide\"\n></amplify-auth-forgot-password-ionic>\n\n<amplify-auth-greetings-ionic\n  *ngIf=\"!shouldHide('Greetings')\"\n  [authState]=\"authState\"\n  [usernameAttributes]=\"_usernameAttributes\"\n></amplify-auth-greetings-ionic>\n\n<amplify-auth-require-new-password-ionic\n  *ngIf=\"!shouldHide('RequireNewPassword')\"\n  [authState]=\"authState\"\n  [hide]=\"hide\"\n></amplify-auth-require-new-password-ionic>\n</div>\n";
var AuthenticatorIonicComponent = /** @class */ (function (_super) {
    __extends(AuthenticatorIonicComponent, _super);
    function AuthenticatorIonicComponent(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    AuthenticatorIonicComponent.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
AuthenticatorIonicComponent.ɵfac = function AuthenticatorIonicComponent_Factory(t) { return new (t || AuthenticatorIonicComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AmplifyService)); };
AuthenticatorIonicComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AuthenticatorIonicComponent, selectors: [["amplify-authenticator-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 18, vars: 7, consts: [[1, "amplify-authenticator", "amplify-authenticator-ionic"], [3, "authState", "usernameAttributes", "hide", 4, "ngIf"], [3, "authState", "signUpConfig", "usernameAttributes", "hide", 4, "ngIf"], [3, "authState", "hide", 4, "ngIf"], [3, "authState", "usernameAttributes", 4, "ngIf"], [3, "authState", "usernameAttributes", "hide"], [3, "authState", "signUpConfig", "usernameAttributes", "hide"], [3, "authState", "hide"], [3, "authState", "usernameAttributes"]], template: function AuthenticatorIonicComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtext(0, "\n");
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵtext(2, "\n\n");
        ɵngcc0.ɵɵtemplate(3, AuthenticatorIonicComponent_amplify_auth_sign_in_ionic_3_Template, 1, 3, "amplify-auth-sign-in-ionic", 1);
        ɵngcc0.ɵɵtext(4, "\n\n");
        ɵngcc0.ɵɵtemplate(5, AuthenticatorIonicComponent_amplify_auth_sign_up_ionic_5_Template, 1, 4, "amplify-auth-sign-up-ionic", 2);
        ɵngcc0.ɵɵtext(6, "\n\n");
        ɵngcc0.ɵɵtemplate(7, AuthenticatorIonicComponent_amplify_auth_confirm_sign_up_ionic_7_Template, 1, 3, "amplify-auth-confirm-sign-up-ionic", 1);
        ɵngcc0.ɵɵtext(8, "\n\n");
        ɵngcc0.ɵɵtemplate(9, AuthenticatorIonicComponent_amplify_auth_confirm_sign_in_ionic_9_Template, 1, 2, "amplify-auth-confirm-sign-in-ionic", 3);
        ɵngcc0.ɵɵtext(10, "\n\n");
        ɵngcc0.ɵɵtemplate(11, AuthenticatorIonicComponent_amplify_auth_forgot_password_ionic_11_Template, 1, 3, "amplify-auth-forgot-password-ionic", 1);
        ɵngcc0.ɵɵtext(12, "\n\n");
        ɵngcc0.ɵɵtemplate(13, AuthenticatorIonicComponent_amplify_auth_greetings_ionic_13_Template, 1, 2, "amplify-auth-greetings-ionic", 4);
        ɵngcc0.ɵɵtext(14, "\n\n");
        ɵngcc0.ɵɵtemplate(15, AuthenticatorIonicComponent_amplify_auth_require_new_password_ionic_15_Template, 1, 2, "amplify-auth-require-new-password-ionic", 3);
        ɵngcc0.ɵɵtext(16, "\n");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtext(17, "\n");
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("SignIn"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("SignUp"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ConfirmSignUp"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ConfirmSignIn"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ForgotPassword"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("Greetings"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("RequireNewPassword"));
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.SignInComponentIonic, ɵngcc4.SignUpComponentIonic, ɵngcc5.ConfirmSignUpComponentIonic, ɵngcc6.ConfirmSignInComponentIonic, ɵngcc7.ForgotPasswordComponentIonic, ɵngcc8.GreetingComponentIonic, ɵngcc9.RequireNewPasswordComponentIonic], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AuthenticatorIonicComponent, [{
        type: Component,
        args: [{
                selector: 'amplify-authenticator-ionic',
                template: template
            }]
    }], function () { return [{ type: ɵngcc1.AmplifyService }]; }, null); })();
    return AuthenticatorIonicComponent;
}(AuthenticatorComponentCore));
export { AuthenticatorIonicComponent };
